src/components/AddNote.jsx

```diff
@@ -6,47 +6,79 @@ import Editor from './Editor'
   }
 
   const addToNotesList = async (note) => {
     if (loggedIn){
-    try {
-      const response = await axios.post('/api/notes', { note }, {
+      try {
+        const response = await axios.post('/api/notes', { note }, {
         headers: {
-          authorization: localStorage.getItem("userToken")
+        authorization: localStorage.getItem("userToken")
         }
-      })
-      if (response.status === 201) {
-       setNotes(response.data.notes)
-       setNote({...note, title: ""})
-       setNote({...note, body: ""})
+        })
+        if (response.status === 201) {
+          setNotesData(data => ({...data, notes: response.data.notes}))
+          setNote(initialNote)
+        }
+      } catch (error) {
+        Toast({type: "error", message:"Oops! Some error occurred."})
+        console.log(error)
       }
-    } catch (error) {
-      console.log(error)
+    } else {
+      Toast({type: "error", message:"Please login"})
     }
-  } else {
-    Toast({type: "error", message:"Please login"})
   }
-}
+
+  const handleSubmit = e => {
+    e.preventDefault();
+    setNote(initialNote)
+  }
+
+  const addTag = (tag) => {
+    tag.length &&
+    setNote(note => ({...note, tags: [...note.tags, tag]}))
+    setTag("")
+  }
 
   return (
     <div className="note new">
-      <div className='note-header'>
-        <input className='note-title' type="text" placeholder='Title' value={note.title} onChange={(e) => setNote({...note, title: e.target.value})}/>     
-      </div>
-      <div className='note-body'>
-        <Editor/>
-      </div>
-      <div className='note-footer'>
-        <ColorPicker changeColor={(color) => handleColor(note, color)}/>
-        <button className='save-note' onClick={() => addToNotesList(note)}><span class="material-icons md-18 material-icons-outlined">add</span></button>
-      </div>
+      <form onSubmit={handleSubmit}>
+        <div className='note-header' >
+          <input className='note-title' type="text" placeholder='Title' value={note.title} onChange={(e) => setNote({...note, title: e.target.value})}/>     
+        </div>
+        <div className='note-body'>
+          <Editor/>
+        </div>
+        <div className='note-footer'>
+          <div className="tags-options">
+            <input type="text" className="tags" placeholder='Add Tags' value={tag} onChange={(e) => setTag(e.target.value)}/>
+            <span class="material-icons md-18 material-icons-outlined add-tag"  onClick={() => addTag(tag)}>add</span>
+            {finalTagsList.map(tag => (
+            <div className='tag-chip'>
+              <p>{tag}</p>
+            </div>
+          ))}
+          </div>
+          <div className='footer-ctas'>
+            <ColorPicker changeColor={(color) => handleColor(note, color)}/>
+            <button className='save-note' type="submit" onClick={()=>addToNotesList(note)}><span class="material-icons md-18 material-icons-outlined">add</span></button>
```

Comment:
Suggestion: you should do this addNote in handle submit

Reply:
Oh yes! I missed changing that, thank you for pointing it out!